src/IECoreDelight/CurvesAlgo.cpp

```diff
@@ -54,7 +54,12 @@ const char *g_bSpline = "b-spline";
 
 void staticParameters( const IECoreScene::CurvesPrimitive *object, ParameterList &parameters )
 {
-	parameters.add( "nvertices", object->verticesPerCurve() );
+	NSIParam_t nvertices = parameters.parameter( "nvertices", object->verticesPerCurve() );
+	nvertices.count = object->verticesPerCurve()->readable().size();
+	nvertices.arraylength = 1;
+	nvertices.flags &= ~NSIParamIsArray;
+
+	parameters.add( nvertices );
```

Comment:
I was hoping to not break the ABI, which I think this would do? Maybe it's not too big of a deal in `GafferDelight`, or is an ABI break in any module an ABI break everywhere? Or maybe I'm just over complicating matters...

Reply:
We can just add an overload with the additional argument, and keep the old function around for a bit before removing it in 1.4. And the implementation for the old function can just forward to the new function, so there's not additional code to maintain.

Reply:
I reverted to using `add()` where we can which I believe means we need the overload in both `add()` and `parameter()`. I made those additions in https://github.com/GafferHQ/gaffer/pull/5432/commits/2941b68c5362fde0c44212533ecc70d6614d9853 and returned to the original, simpler methods for primitive data handling.